service/frontend/workflow_handler.go

```diff
@@ -4121,6 +4113,46 @@ func (wh *WorkflowHandler) validateWorkflowIdReusePolicy(
 	return nil
 }
 
+func (wh *WorkflowHandler) validateWorkflowCompletionCallbacks(
+	ns namespace.Name,
+	callbacks []*commonpb.Callback,
+) error {
+	if len(callbacks) > 0 && !wh.config.EnableCallbackAttachment(ns.String()) {
+		return status.Error(
+			codes.InvalidArgument,
+			"attaching workflow callbacks is disabled for this namespace",
+		)
+	}
+
+	for _, callback := range callbacks {
+		switch cb := callback.GetVariant().(type) {
+		case *commonpb.Callback_Nexus_:
+			if len(cb.Nexus.GetUrl()) > wh.config.CallbackURLMaxLength() {
+				return status.Error(
+					codes.InvalidArgument,
+					fmt.Sprintf(
+						"invalid url: url length longer than max length allowed of %d",
+						wh.config.CallbackURLMaxLength(),
+					),
+				)
+			}
+			u, err := url.Parse(cb.Nexus.GetUrl())
+			if err != nil {
+				return status.Errorf(codes.InvalidArgument, "invalid url: %v", err)
+			}
+			if !(u.Scheme == "http" || u.Scheme == "https") {
+				return status.Errorf(codes.InvalidArgument, "invalid url scheme for url: %v", u)
```

Comment:
```suggestion
				return status.Errorf(codes.InvalidArgument, "invalid url: unknown scheme %v", u)
```
				
This keeps it consistent with the other errors